addons/web/static/tests/views/list_tests.js

```diff
@@ -1978,14 +1926,45 @@ QUnit.module('Views', {
         });
 
         await testUtils.dom.click(form.$('.o_field_boolean input'));
+
+        assert.strictEqual(form.el.querySelector('th').style.width, "",
+            "Column header should be initially unfrozen");
+
         await testUtils.dom.click(form.$('.nav-item:last() .nav-link'));
 
-        assert.strictEqual(form.$('th')[0].style.width, "",
-            "Column header should be reset and remain unfrozen");
+        assert.notEqual(form.el.querySelector('th').style.width, "",
+            "Column header should have been frozen");
 
         form.destroy();
     });
 
+    QUnit.test('empty list: state with nameless and stringless buttons', async function (assert) {
+        assert.expect(3);
+
+        this.data.foo.records = [];
+        const list = await createView({
+            arch: `
+            <tree>
+                <field name="foo"/>
+                <field name="bar"/>
+                <button string="choucroute"/>
+                <button icon="fa-heart"/>
+            </tree>`,
+            data: this.data,
+            model: 'foo',
+            View: ListView,
+        });
+
+        assert.notEqual(list.el.querySelector('th[data-name="foo"]').style.width, '25%',
+            "Field column should be frozen");
+        assert.notEqual(list.el.querySelector('th[data-string="choucroute"]').style.width, '25%',
+            "Nameless button column should be frozen");
+        assert.notEqual(list.el.querySelector('th[data-icon="fa-heart"]').style.width, '25%',
+            "Nameless and stringless button should be frozen");
```

Comment:
not sure to understand this test

Reply:
It actually has 2 purposes: assert that a `data-name|string|icon` is correctly assigned to each column, and that they all get to be frozen (`_computeDefaultWidths` works with column data-attributes and wouldn't work on column that don't have one). Initially, there were assertions to simply verify that the attributes were given, but it's already done by just selected the column headers. Finally, we don't check the boolean value (bar) since it also has a ` data-name`.

Reply:
as discussed, maybe check instead that the width is expressed in px